packages/core/src/ObservableActorRef.ts

```diff
@@ -1,41 +1,47 @@
 import { EventObject, ActorRef, Behavior, SCXML, ActorContext } from './types';
-import { Actor } from './actor';
 import { symbolObservable, toSCXMLEvent } from './utils';
+import { Mailbox } from './Mailbox';
+import { LifecycleSignal, startSignal, stopSignal } from './actors';
+
+const nullSubscription = {
+  unsubscribe: () => void 0
+};
 
 export class ObservableActorRef<TEvent extends EventObject, TEmitted>
   implements ActorRef<TEvent, TEmitted> {
   private current: TEmitted;
   public deferred = true;
   private context: ActorContext<TEvent, TEmitted>;
-  private actor: Actor<TEvent, TEmitted>;
   public name: string;
+  private mailbox = new Mailbox(this._process.bind(this));
 
-  constructor(behavior: Behavior<TEvent, TEmitted>, name: string) {
+  constructor(public behavior: Behavior<TEvent, TEmitted>, name: string) {
```

Comment:
q: why `behavior` is `public`? shouldn't it be hidden from the consumer?

Reply:
Fixed in [ac21c9f](https://github.com/statelyai/xstate/pull/3148/commits/ac21c9fe18e3ff3c77dc1c7ffe0979255fe5b4ad)